### PR TITLE
Some Vagrantfile updates

### PR DESCRIPTION
**Pull request by bjackman**
Hi, I'm not sure if this is of much value. I ended up discovering that my distro packages Nix and abandoning the use of Vagrant, so this has barely been tested, I just built the code once in the new boxen. Nonetheless maybe it saves someone some time down the line...?

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -69,13 +70,14 @@ Vagrant.configure("2") do |config|
       'scripts'        => [ $ubuntu_deps, ],
       'fix_console'    => 0,
     },
-    'ubuntu-21.04'     => {
-      'image'          => 'ubuntu/hirsute64',
+    'ubuntu-22.04'     => {
```
Comment (bjackman) [inline Vagrantfile:73]: Oh actually I forgot `llvm.sh` doesn't work on 22.04. So might as well just delete this.

**Comment by viktormalik** [comment]
Thanks for opening this!

Even though adding Ubuntu 22.04 may not work, I think that it would be nice to do something about our Vagrant support - either fix it to some minimal working state or delete it completely.

**Comment by fbs** [comment]
My main reason for vagrant was that a few years ago it was the easiest way to do bpftrace dev on macos. Nowdays `lima` might be a better tool (e.g. https://github.com/lima-vm/lima/blob/master/examples/ubuntu.yaml ). Or maybe even with the various docker engines as they all run in VMs anyway

**Comment by bjackman** [comment]
Yeah, I did also wonder if a Dockerfile/Containerfile would make Vagrant
irrelevant here. It seems it might be easier to make it hermetic in the
ways that matter (software dependencies), and you can avoid the pointless
hermeticism that Vagrant gives you (fixed CPUs/RAM in the guest).

**Comment by danobi** [comment]
Yeah vagrant may not be the best thing to support going forwards. I recall I spent around a day trying to get vagrant to work on an M1 mac and it just didn't. None of the providers supported M1, at least ~6 months ago. Not sure what current status is

**Comment by viktormalik** [comment]
Agreed, supporting Vagrant with different providers is difficult. My suggestion would be to remove `Vagrantfile` completely.

**Comment by danobi** [comment]
Ok, let's go with deleting vagrant then

**Pull request closed without merging**